test/ip_constexpr_test.cpp

```diff
@@ -0,0 +1,76 @@
+//
+// ip_constexpr_test.cpp
+//
+// Copyright 2025 Mathias Stearn
+//
+// Distributed under the Boost Software License, Version 1.0.
+// See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt
+//
+
+#include <boost/intrusive_ptr.hpp>
+
+#ifndef BOOST_SP_NO_CXX20_CONSTEXPR
+
+struct dummy {
+    // no-ops, so safe on pointers to static constexpr variables
+    friend constexpr void intrusive_ptr_add_ref(const dummy *) {}
+    friend constexpr void intrusive_ptr_release(const dummy *) {}
+};
+static constexpr dummy d;
+
+struct subdummy : dummy {};
+
+// Test that basic operations work at compile time.
+static_assert(bool(boost::intrusive_ptr(&d)));
+static_assert(!bool(boost::intrusive_ptr<dummy>(nullptr)));
+static_assert(!bool(boost::intrusive_ptr<dummy>()));
+static_assert(!bool(boost::intrusive_ptr<dummy>(boost::intrusive_ptr<subdummy>())));
+static_assert(&*boost::intrusive_ptr(&d));
```

Comment:
The results of the operations should be tested too; `&*boost::intrusive_ptr<dummy>(&d) == &d`

Reply:
Since we're testing the result in either case (against nullptr), we might as well test it for the right value.

At the moment this makes no difference but only because we know that the implementation doesn't do `if consteval` anywhere. Tests should not assume knowledge of the implementation.